dScripts/AgSurvivalBuffStation.cpp

```diff
@@ -1,15 +1,43 @@
 #include "AgSurvivalBuffStation.h"
+#include "DestroyableComponent.h"
+#include "EntityManager.h"
+#include "GameMessages.h"
 #include "SkillComponent.h"
 #include "dLogger.h"
 
 void AgSurvivalBuffStation::OnRebuildComplete(Entity* self, Entity* target) {
-    auto skillComponent = self->GetComponent<SkillComponent>();
+    auto destroyableComponent = self->GetComponent<DestroyableComponent>();
+    // We set the faction to 6 so that the buff station sees players as friendly targets to buff
+    if (destroyableComponent != nullptr) destroyableComponent->SetFaction(6);
 
-    if (skillComponent == nullptr) return;
+    auto skillComponent = self->GetComponent<SkillComponent>();
 
-    skillComponent->CalculateBehavior(201, 1784, self->GetObjectID());
+    if (skillComponent != nullptr) skillComponent->CalculateBehavior(skillIdForBuffStation, behaviorIdForBuffStation, self->GetObjectID());
 
-    self->AddCallbackTimer(10.0f, [self]() {
+    self->AddCallbackTimer(smashTimer, [self]() {
         self->Smash();
     });
+    self->AddTimer("DropArmor", dropArmorTimer);
+    self->AddTimer("DropLife", dropLifeTimer);
+    self->AddTimer("Dropimagination", dropImaginationTimer);
+    self->SetVar<LWOOBJID>(u"PlayerId", target->GetObjectID());
+}
+
+void AgSurvivalBuffStation::OnTimerDone(Entity* self, std::string timerName) {
+    auto targetID = self->GetVar<LWOOBJID>(u"PlayerId");
+    auto target = EntityManager::Instance()->GetEntity(targetID);
+    uint32_t powerupToDrop = lifePowerup;
+    if (timerName == "DropArmor") {
+        powerupToDrop = armorPowerup;
+        self->AddTimer("DropArmor", dropArmorTimer);
+    }
+    if (timerName == "DropLife") {
+        powerupToDrop = lifePowerup;
+        self->AddTimer("DropLife", dropLifeTimer);
+    }
+    if (timerName == "Dropimagination") {
+        powerupToDrop = imaginationPowerup;
+        self->AddTimer("Dropimagination", dropImaginationTimer);
+    }
+    if (target != nullptr) GameMessages::SendDropClientLoot(target, self->GetObjectID(), powerupToDrop, 0, self->GetPosition());
```

Comment:
Does this only create powerups for the player who built the Survival Buff Station? If so, then it might not be accurate to live, see https://youtu.be/iWe3KZv2TJE?t=92. The Survival Buff Station was built by a team member ([here](https://youtu.be/iWe3KZv2TJE?t=66)) but the powerups are available for all team members. Unfortunately I don't know what the exact correct behavior is; whether powerups appeared for every player, or whether they were "shared" somehow, so one player could steal them from another player.

I think the same also applied to the powerup spawners; but that is probably off-topic here and rather related to #344.

Reply:
Second part is a separate issue.  First part is not apart of the original script.  Original script only sends the buffs to the player that built it.

Reply:
Updated script to drop loot buffs for all members in a team.